File: backend/app/schemas/public.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class LandingResponse(BaseModel):
    business_name: str
    location_name: str
    host_name: str | None = None
    landing_title: str | None = None
    welcome_message: str | None = None
    suggested_questions: list[str] = Field(default_factory=list)
    primary_color: str | None = None
    secondary_color: str | None = None


class ChatMessageRequest(BaseModel):
    message: str = Field(min_length=1, max_length=4000)


class ChatMessageResponse(BaseModel):
    id: uuid.UUID
    role: str
    content: str
    created_at: datetime

    model_config = {"from_attributes": True}


class SendMessageResponse(BaseModel):
    conversation_id: uuid.UUID
    reply: ChatMessageResponse


class ConversationHistoryResponse(BaseModel):
    conversation_id: uuid.UUID | None = None
    messages: list[ChatMessageResponse]

File: frontend/next/src/customer/api/public.ts
import { api } from "./client";

export interface LandingResponse {
  business_name: string;
  location_name: string;
  host_name: string | null;
  landing_title: string | null;
  welcome_message: string | null;
  suggested_questions: string[];
  primary_color: string | null;
  secondary_color: string | null;
}

export interface ChatMessageResponse {
  id: string;
  role: string;
  content: string;
  created_at: string;
}

export interface SendMessageResponse {
  conversation_id: string;
  reply: ChatMessageResponse;
}

export interface ConversationHistoryResponse {
  conversation_id: string | null;
  messages: ChatMessageResponse[];
}

export function fetchLanding(businessSlug: string, locationSlug: string, sessionToken?: string) {
  return api.get<LandingResponse>(
    `/api/public/${businessSlug}/${locationSlug}/landing`,
    sessionToken,
  );
}

export function sendMessage(
  businessSlug: string,
  locationSlug: string,
  sessionToken: string,
  message: string,
) {
  return api.post<SendMessageResponse>(
    `/api/public/${businessSlug}/${locationSlug}/messages`,
    { message },
    sessionToken,
  );
}

export function fetchConversation(
  businessSlug: string,
  locationSlug: string,
  sessionToken: string,
) {
  return api.get<ConversationHistoryResponse>(
    `/api/public/${businessSlug}/${locationSlug}/conversation`,
    sessionToken,
  );
}
